### PR TITLE
[Merged by Bors] - feat: port Algebra.MonoidAlgebra.Ideal

### PR DESCRIPTION
**Pull request by newell**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/MonoidAlgebra/Ideal.lean [hunk 1]
```diff
@@ -28,31 +28,34 @@ theorem MonoidAlgebra.mem_ideal_span_of_image [Monoid G] [Semiring k] {s : Set G
...
+    refine Ideal.span_le.2 ?_
     rintro _ ⟨i, hi, rfl⟩ m hm
     refine' ⟨_, hi, 1, _⟩
-    obtain rfl := finset.mem_singleton.mp (Finsupp.support_single_subset hm)
+    obtain rfl := Finset.mem_singleton.mp (Finsupp.support_single_subset hm)
     exact (one_mul _).symm
   · intro hx
     rw [← Finsupp.sum_single x]
-    apply Ideal.sum_mem _ fun i hi => _
+    refine Ideal.sum_mem _ fun i hi => ?_  -- porting note: changed `apply` to `refine`
```
Comment (eric-wieser) [inline Mathlib/Algebra/MonoidAlgebra/Ideal.lean]: For reviewers: this failure is really wierd, Lean totally ignores the `fun` and doesn't actually introduce `i` or `hi` if I use `apply`

File: Mathlib/Algebra/MonoidAlgebra/Ideal.lean [hunk 2]
```diff
@@ -28,31 +28,34 @@ theorem MonoidAlgebra.mem_ideal_span_of_image [Monoid G] [Semiring k] {s : Set G
...
+        obtain ⟨xm, -, hm⟩ := hm
+        replace hm := Finset.mem_bunionᵢ.mp (Finsupp.support_sum hm)
         obtain ⟨ym, hym, hm⟩ := hm
-        replace hm := finset.mem_singleton.mp (Finsupp.support_single_subset hm)
+        replace hm := Finset.mem_singleton.mp (Finsupp.support_single_subset hm)
         obtain rfl := hm
-        refine' (hy _ hym).imp fun sm => Exists.imp fun hsm => _
+        -- porting note: changed `Exists.imp` to `And.imp_right` due to change in `∃ x ∈ s`
+        -- elaboration
+        refine' (hy _ hym).imp fun sm p => And.imp_right _ p
```
Comment (eric-wieser) [inline Mathlib/Algebra/MonoidAlgebra/Ideal.lean]: Are we expected to change the proofs, or fix the not-matching-lean-3 mathport output?

@digama0, should this be `#align`ed with a subscript `x` if the latter?

File: Mathlib/Algebra/MonoidAlgebra/Ideal.lean [hunk 3]
```diff
@@ -28,31 +28,34 @@ theorem MonoidAlgebra.mem_ideal_span_of_image [Monoid G] [Semiring k] {s : Set G
...
         rintro ⟨d, rfl⟩
         exact ⟨xm * d, (mul_assoc _ _ _).symm⟩ }
   change _ ↔ x ∈ RHS
   constructor
   · revert x
-    refine' Ideal.span_le.2 _
+    rw [←SetLike.le_def] -- porting note: refine needs this even though it's defeq?
```
Comment (eric-wieser) [inline Mathlib/Algebra/MonoidAlgebra/Ideal.lean:50]: This one is weird too, and possibly worth a zulip thread.

File: Mathlib/Algebra/MonoidAlgebra/Ideal.lean (hunk 2 above)
Comment (jcommelin) [reply Mathlib/Algebra/MonoidAlgebra/Ideal.lean]: It seems to me that in practice we just change the proofs and move on.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**